Volume/Supporting/UserData.swift

```diff
@@ -120,12 +120,31 @@ class UserData: ObservableObject {
     }
 
     func set(publication: Publication, isFollowed: Bool) {
+        guard let uuid = UserDefaults.standard.string(forKey: userUUIDKey) else {
+            print("Error: failed to set follow status because user UUID not set")
+            return
+        }
+        
         if isFollowed {
-            if !followedPublicationIDs.contains(publication.id) {
-                followedPublicationIDs.insert(publication.id, at: 0)
-            }
+            cancellables[.follow(publication)] = Network.shared.publisher(for: FollowPublicationMutation(publicationID: publication.id, uuid: uuid))
+                .sink { completion in
+                    if case let .failure(error) = completion {
+                        print(error)
+                    }
+                } receiveValue: { value in
+                    if !self.followedPublicationIDs.contains(publication.id) {
+                        self.followedPublicationIDs.insert(publication.id, at: 0)
+                    }
```

Comment:
I've got my 4410 hat on: This isn't thread safe. What if you spam the follow button. Is it possible that you end up seeing the same publication id twice? Probably not in reality, but still.